### PR TITLE
Introduced safer file checking. 

### PR DESCRIPTION
**Pull request by cjke**
Changed argument formatting to be the more standard -- pattern as used in typical shell commands
Added argument defaults and aliases to simplify argument logic. File checks added to all file writes.

### CODE REVIEW

**Comment by cjke** [comment]
Addresses #4 

**Note** this introduces breaking changes (if this tool is being used in any build flows, the new flags would be a breaking change). Semver will need to account for this. 

**Edit**

Also want to mention that (imo) the dir flag should either be an actual flag (like --dir or --process-as-dir), or alternatively, have two scripts, one for single, and one for multiple (the whole unix philosophy: Make each program do one thing well. To do a new job, build afresh rather than complicate old programs by adding new "features".) Your core script would still be shared among both scripts. If this sounds like something of interest, I will raise it as a new issue.

**Comment by goopscoop** [comment]
This looks great. Will test and merge. Also, I'm all for the changes you propose for separating the --dir into its own script.

**Comment by goopscoop** [comment]
I ran `svgtoreact dir` and the err message isn't showing up. It's still appears to be overriding the current js files.

**Comment by cjke** [comment]
Hmm, thats odd - perhaps I am not using dir as you intended, when I perform the following commands, it generates the errors:

image

*Notes* 
 - State 1 - directory has two svg's, no components
 - State 2 - after first run, two svg's, two components
 - State 3 - after second run, two svg's, and error message
 - State 4 - remove one component, rerun, one error message, one new component (plus existing)

**Comment by goopscoop** [comment]
Yeah, that appears to be working correctly on your machine. Here's what I'm seeing after pulling your branch, installing and running it.

1__codyb_15mbp-09924____home_rougish_src_assets__zsh_

File: index.js [hunk 1]
```diff
@@ -354,10 +371,35 @@ const runUtilForAllInDir = () => {
...
 };
 
+/**
+ * Tests if a file exists
+ * @param  string file The path to the file being read
+ * @return boolean Boolean indicating if the file exists
+ */
+function fileExists(file) {
+  try {
+    fs.accessSync(file, fs.constants.R_OK | fs.constants.W_OK);
```
Comment (goopscoop) [inline index.js:381]: Appears to be failing on this line. 

```
  console.log(fs.accessSync(file, fs.constants.R_OK || fs.constants.W_OK))
                                              ^

TypeError: Cannot read property 'R_OK' of undefined
    at fileExists (/Users/codyb/Home/svg-to-react/index.js:380:47)
    at writeFile (/Users/codyb/Home/svg-to-react/index.js:306:6)
    at /Users/codyb/Home/svg-to-react/index.js:337:5
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)
```

Comment (cjke) [reply index.js:381]: Gotcha, makes a lot more sense - between node 6.2 and 6.3 there is fs.constants.R_OK and fs.R_OK. Poor assumption on my part to assuem >= 6.3. Closing this pull request, and I will resubmit taking into account the different node versions. What is your support level (node-wise) for this project?

Comment (goopscoop) [reply index.js:381]: Sorry for the delay. Had a busy week with family and stuff. Anyway, I'm thinking we support node v >= 4.6. Any thoughts?

Comment (cjke) [reply index.js:381]: Ok no worries, I will tweak a few things an submit a new PR

Comment (goopscoop) [reply index.js:381]: Cool. Sounds good. I appreciate it.

**Pull request closed without merging**